### PR TITLE
Test fixtures for CliRunner, HOME, and XDG_*

### PR DESCRIPTION
**Pull request by tiran**
Introduce pytest fixtures for click's `CliRunner`, to reset `HOME` and `XDG_*` env variables, and access to test data.

`tmp_path_home` fixture sets `os.environ` variable `HOME` to pytest's current `tmp_path` and removes all `XDG_*` variables.

`cli_runner` uses `tmp_path_home` and creates a new click `CliRunner` with file system isolation inside pytest's `tmp_path`.

`testdata_path` returns the local test data directory as absolute `pathlib.Path` object.

All tests have been refactored to use the new `cli_runner` fixture. The change is inspired by a previous PR by Ihar Hrachyshka.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by tiran** [comment]
The PR solves some problems for PR #1471 around handling of `HOME` and XDG directories in tests.

File: tox.ini [hunk 1]
```diff
@@ -24,7 +24,6 @@ deps =
     pytest-html
 commands =
     ilab --version
-    {envpython} -m instructlab --version
```
[the remark this answers is not in the document]
Comment (tiran) [reply tox.ini:27]: I removed it because it was slowing down every test run. I can re-add it.

[the remark this answers is not in the document]
Comment (tiran) [reply tox.ini:27]: The check is now in `functional-tests.sh`.

[the remark this answers is not in the document]
Comment (leseb) [reply tox.ini:27]: Makes more sense!

**Review by RobotSail — APPROVED** [review]
LGTM

**Review by leseb — APPROVED** [review]

**Pull request merged**